Core/Security/m2crypto/X509Chain.py

```diff
@@ -252,6 +252,9 @@ def __certListFromPemString(certString):
     # To get list of X509 certificates (not X509 Certificate Chain) from string it has to be parsed like that
     # (constructors are not able to deal with big string)
     certList = []
+    # If the certificate is downloaded from the server it will be a str in Python 3
+    if six.PY3 and isinstance(certString, six.string_types):
```

Comment:
I understand and agree that this needs to be done, but should it really be hidden there, or should we rather do it upon reception by the server ?

Reply:
I'm not sure where it can be better put as it would have to be fixed wherever RPC calls are made to `Framework/ProxyManager`, for which there is at least 4 places in `ProxyManagerClient`. Plus if you get it wrong the error gets lost in DIRAC's error handling making it a pain to debug.

As it stands this is still technically "correct" as the string is a still a valid representation of the certificate chain. In other Python 3 libraries it's quite common for interfaces like this to be permissive and accept both `bytes` and `str` where it can be done safely and without ambiguities.

Reply:
> upon reception by the server

For context, this crashes on the client when running the LHCb "submit jobs as different users" test where a admin user is explicitly downloading data from the ProxyDB.

Reply:
ok then good enough for me :-)